### ISSUE TITLE
Many examples use `HWCN` order instead of `WHCN` recommended by document

### ISSUE THREAD

**Issue by yiyuezhuo** [FluxML/model-zoo#235]
According to [doc](https://fluxml.ai/Flux.jl/stable/models/layers/#Convolution-and-Pooling-Layers-1)

> Data should be stored in WHCN order (width, height, # channels, batch size). In other words, a 100×100 RGB image would be a 100×100×3×1 array, and a batch of 50 would be a 100×100×3×50 array.

But many examples use `HWCN`, for [example](https://github.com/FluxML/model-zoo/blob/master/vision/cifar10/cifar10.jl#L25):

```julia
# Function to convert the RGB image to Float64 Arrays
function getarray(X)
    Float32.(permutedims(channelview(X), (2, 3, 1)))
end
```

The correct transform should be `Float32.(permutedims(channelview(X), (3, 2, 1)))`, because channelview(X) returns a "CHW" array. Likewise, MNIST [example](https://github.com/FluxML/model-zoo/blob/master/vision/mnist/conv.jl#L32) doesn't use any `permutedims` so it just keeps its wrong "HW" order. Fortunately, some are correct, for example, this [one](https://github.com/FluxML/model-zoo/blob/master/vision/vae_mnist/vae_mnist.jl#L24) from `MLDatasets`. The three cases are shown in this [gist](https://gist.github.com/yiyuezhuo/ff997441a773525489a499eede273f8e).

So basically, many examples in fact run models on a "transposed" image datasets. The good (bad?) part is that CNN is robust enough to deal with this distortion, so we can't detect it from statistics such as acc and even eye evaluation. But those examples suggest a misleading preprocessing pipeline and should be fixed. (To be honest, I post this issue since I have been misleaded by this...)

**Comment by aditkumar72** [comment]
Fixed in #306

**Comment by MariusDrulea** [comment]
I think FluxML shall also document why we use the WHCN order. The only explanation I have is that CUDA.jl is using cuDNN and cuDNN supports the NCHW order (row-major). If we look at the NCHW memory representation in cuDNN, we can notice this is exactly the same as the memory representation of WHCN in julia.

**cuDNN:**
https://docs.nvidia.com/deeplearning/cudnn/developer-guide/index.html#nchw-layout-x32
![image](https://user-images.githubusercontent.com/27725572/219868293-dec56740-6dd1-41fe-918e-d6611018c3af.png)

![image](https://user-images.githubusercontent.com/27725572/219872777-73da680a-2258-442e-ab24-604a488b63c7.png)
